File: typescript/src/lib/random.ts
const alphanumericChars: string =
  "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
const alphaChars: string =
  "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";

function random(length: number = 16, alphanumeric: boolean = false): string {
  let result: string = alphaChars.charAt(
    Math.floor(Math.random() * alphaChars.length)
  );

  let chars: string;

  if (alphanumeric) {
    chars = alphanumericChars;
  } else {
    chars = alphaChars;
  }

  for (let i: number = 1; i < length; i++) {
    const randomIndex: number = Math.floor(Math.random() * chars.length);
    result += chars.charAt(randomIndex);
  }

  return result;
}

export { random };

File: src/nucleoid/lib/random.py
"""Random string generation utilities"""

import random as py_random
import string

# Character sets for random string generation
ALPHANUMERIC_CHARS = string.ascii_letters + string.digits
ALPHA_CHARS = string.ascii_letters


def random(length: int = 16, alphanumeric: bool = False) -> str:
    """
    Generate a random string of specified length.

    Args:
        length: Length of the random string (default: 16)
        alphanumeric: If True, include digits; if False, only letters (default: False)

    Returns:
        A random string starting with a letter, followed by random characters
    """
    # First character is always a letter
    result = py_random.choice(ALPHA_CHARS)

    # Choose character set based on alphanumeric flag
    chars = ALPHANUMERIC_CHARS if alphanumeric else ALPHA_CHARS

    # Generate remaining characters
    for _ in range(1, length):
        result += py_random.choice(chars)

    return result
